### PR TITLE
Read subscription ID from instance metadata

### PR DESCRIPTION
**Pull request by fiunchinho**
The same way that we don't have to explicitly pass credentials as arguments when using managed identity extension, we could allow people to avoid passing the `subscriptionID` if they don't want to, and fetch it from the instance metadata.

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/azure/azure_manager.go [hunk 1]
```diff
@@ -26,9 +26,11 @@ import (
 
 	"github.com/Azure/go-autorest/autorest/azure"
 	"gopkg.in/gcfg.v1"
+	"k8s.io/klog"
+	azure2 "k8s.io/legacy-cloud-providers/azure"
```
Comment (feiskyer) [inline cluster-autoscaler/cloudprovider/azure/azure_manager.go]: looks good to me in general. Could we rename azure2 to providerazure?

Comment (feiskyer) [reply cluster-autoscaler/cloudprovider/azure/azure_manager.go]: Also, please group k8s.io packages togather

Comment (fiunchinho) [reply cluster-autoscaler/cloudprovider/azure/azure_manager.go]: Fixed. I also added a note to explain the new behavior. Please take a look.

**Comment by fiunchinho** [comment]
@feiskyer the build has failed, but I can't build the project locally. It seems like there is a problem with an outdated dependency? Can you help me with this?

**Comment by fiunchinho** [comment]
I believe I fixed it.

File: cluster-autoscaler/cloudprovider/azure/azure_manager.go [hunk 2]
```diff
@@ -534,3 +541,22 @@ func parseLabelAutoDiscoverySpec(spec string) (labelAutoDiscoveryConfig, error)
...
+// getSubscriptionId reads the Subscription ID from the instance metadata.
+func getSubscriptionIdFromInstanceMetadata() (string, error) {
+	subscriptionID, present := os.LookupEnv("ARM_SUBSCRIPTION_ID")
+	if !present {
+		metadataService, err := providerazure.NewInstanceMetadataService(metadataURL)
+		if err != nil {
+			return "", err
+		}
+
+		metadata, err := metadataService.GetMetadata(0)
```
Comment (feiskyer) [inline cluster-autoscaler/cloudprovider/azure/azure_manager.go]: why `(0)` is passed in here?

Comment (feiskyer) [reply cluster-autoscaler/cloudprovider/azure/azure_manager.go]: I see, a new param is added now

**Pull request merged**